spec/ScoreCardSpec.js

```diff
@@ -0,0 +1,159 @@
+describe("ScoreCard", function() {
+  var scorecard;
+
+  beforeEach(function() {
+    scorecard = new ScoreCard();
+  });
+
+  it("should keep track of the frame", function() {
+    expect(scorecard.frame).toEqual(1);
+  });
+
+  it("should know there are 10 bowling pins at the start", function() {
+    expect(scorecard.pins).toEqual(10);
+  });
+
+  it("can return the total for each roll", function() {
+    expect(scorecard.scoreForRoll(5)).toEqual(5);
+  });
+
+  it("can move to the next frame", function() {
+    scorecard.nextFrame();
+    expect(scorecard.frame).toEqual(2);
+  });
+
+  it("is limited to 11 frames", function() {
+    for(var i = 0; i < 11; i++) {
+      scorecard.nextFrame();
+    }
+    expect(scorecard.nextFrame()).toEqual("End of game!");
+  });
+
+  it("can move to the next roll", function() {
+    scorecard.nextRoll();
+    expect(scorecard.roll).toEqual(2);
+  });
+
+  it("can return the points for a frame", function() {
+    scorecard.scoreForRoll(4);
+    scorecard.scoreForRoll(3);
+    expect(scorecard.scoreForFrame(1)).toEqual(7);
+  });
+
+  it("adds points to scores each roll", function() {
+    scorecard.scoreForRoll(4);
+    scorecard.scoreForRoll(5);
+    expect(scorecard.scores[1]).toEqual([4, 5]);
```

Comment:
This, however, is breaking encapsulation: we're reaching into scorecard to extract a particular value (an array) from its scores array. This would be much better done by an interface as you've done above (line 37).